consensus/polybft/fsm.go

```diff
@@ -467,11 +509,30 @@ func (f *fsm) VerifyStateTransactions(transactions []*types.Transaction) error {
 			if err := f.verifyDistributeRewardsTx(tx); err != nil {
 				return fmt.Errorf("error while verifying distribute rewards transaction. error: %w", err)
 			}
+		case *contractsapi.SlashValidatorSetFn:
+			if !forkmanager.GetInstance().IsForkEnabled(chain.DoubleSignSlashing, f.Height()) {
+				return errSlashingTxNotExpected
+			}
+
+			if slashingTxExists {
+				return errSlashingTxSingleExpected
+			}
+
+			slashingTxExists = true
+
+			if err := f.verifySlashingTx(tx); err != nil {
+				return fmt.Errorf("error while verifying slashing transaction. error: %w", err)
```

Comment:
Nitpick (consider if we can better format error message)
```suggestion
				return fmt.Errorf("error while verifying slashing transaction: %w", err)
```

Reply:
Fixed [3a65efb](https://github.com/0xPolygon/polygon-edge/pull/1815/commits/3a65efb21cb39f9cc4f50c963d86c19959e114dd)